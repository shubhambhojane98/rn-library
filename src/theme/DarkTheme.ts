import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
