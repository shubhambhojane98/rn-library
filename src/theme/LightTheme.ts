import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
