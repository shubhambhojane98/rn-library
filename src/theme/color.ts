import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
