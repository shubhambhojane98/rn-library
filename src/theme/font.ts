import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }