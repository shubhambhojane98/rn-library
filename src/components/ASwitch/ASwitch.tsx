import React, {useEffect} from 'react';
import {Animated, Pressable, StyleSheet, I18nManager, View} from 'react-native';
import {Color} from '../../theme';

interface SwitchProps {
  onValueChange: () => void;
  switchSize: number;
  isActive: boolean;
  activeColor?: string;
  inActiveColor?: string;
  knobColor?: string;
  isDisabled: boolean;
}

const ASwitch: React.FC<SwitchProps> = ({
  onValueChange,
  switchSize,
  isActive,
  activeColor = Color.switchActiveBgColor,
  inActiveColor = Color.switchInActiveBgColor,
  knobColor = Color.switchKnobColor,
  isDisabled = false,
}) => {
  const trackWidth = React.useMemo(() => {
    return switchSize * 2;
  }, [switchSize]);
  const trackHeight = React.useMemo(() => {
    return switchSize;
  }, [switchSize]);
  const knobSize = React.useMemo(() => {
    return switchSize * 0.6;
  }, [switchSize]);

  const knobAnim = React.useRef(new Animated.Value(0)).current;
  const knobPos = I18nManager.isRTL
    ? isActive
      ? -trackWidth / 4
      : trackWidth / 4
    : isActive
    ? trackWidth / 4
    : -trackWidth / 4;

  const trackAnim = React.useRef(new Animated.Value(0)).current;
  const trackColor = isActive ? 75 : -75;

  const interpolatedColorAnimation = trackAnim.interpolate({
    inputRange: [-75, 75],
    outputRange: [inActiveColor, activeColor],
  });

  useEffect(() => {
    Animated.parallel([
      Animated.spring(knobAnim, {
        toValue: knobPos,
        useNativeDriver: false,
      }),
      Animated.timing(trackAnim, {
        toValue: trackColor,
        duration: 200,
        useNativeDriver: false,
      }),
    ]).start();
  }, [isActive]);

  const handleSwitch = () => {
    if (isDisabled) {
      return;
    }
    onValueChange();
  };

  return (
    <Pressable onPress={handleSwitch} style={{width: switchSize}}>
      <View style={styles({switchSize}).container}>
        <Animated.View
          style={[
            styles({switchSize}).trackView,
            {
              width: trackWidth,
              height: trackHeight,
              borderRadius: trackHeight / 2,
              borderColor: isDisabled ? Color.greyOpac50 : Color.black,
              backgroundColor: isDisabled
                ? Color.greyOpac10
                : interpolatedColorAnimation,
            },
          ]}
        />
        <Animated.View
          style={[
            {
              width: switchSize,
              height: switchSize,
              borderRadius: switchSize / 2,
              transform: [{translateX: knobAnim}],
            },
            styles({switchSize}).knobContainer,
          ]}>
          <Animated.View
            style={{
              width: knobSize,
              height: knobSize,
              borderRadius: knobSize / 2,
              backgroundColor: isDisabled ? Color.greyOpac50 : knobColor,
            }}
          />
        </Animated.View>
      </View>
    </Pressable>
  );
};

export default ASwitch;

const styles = (props: {switchSize: number}) =>
  StyleSheet.create({
    container: {
      alignItems: 'center',
      justifyContent: 'center',
    },
    trackView: {
      position: 'absolute',
      borderWidth: 1,
    },
    knobContainer: {
      alignItems: 'center',
      justifyContent: 'center',
    },
  });
