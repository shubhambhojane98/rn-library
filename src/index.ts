import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
