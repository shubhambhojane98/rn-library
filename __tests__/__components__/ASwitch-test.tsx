import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ASwitch from '../../src/components/ASwitch/ASwitch';
import {Color} from '../../src/theme';

it('renders correctly default prop', () => {
  const tree = renderer
    .create(
      <ASwitch
        onValueChange={jest.fn()}
        switchSize={30}
        isActive={true}
        isDisabled={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with disable', () => {
  const tree = renderer
    .create(
      <ASwitch
        onValueChange={jest.fn()}
        switchSize={30}
        isActive={false}
        isDisabled={true}
        activeColor={Color.environment}
        inActiveColor={Color.yellow}
        knobColor={Color.red}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly activeColor & inactive & knobColor  prop', () => {
  const tree = renderer
    .create(
      <ASwitch
        onValueChange={jest.fn()}
        switchSize={30}
        isActive={false}
        isDisabled={false}
        activeColor={Color.environment}
        inActiveColor={Color.yellow}
        knobColor={Color.red}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
