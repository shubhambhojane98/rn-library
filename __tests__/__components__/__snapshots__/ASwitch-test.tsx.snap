// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly activeColor & inactive & knobColor  prop 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "width": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(125, 181, 35, 1)",
          "borderColor": "#231F20",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 30,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#A81E23",
            "borderRadius": 9,
            "height": 18,
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly default prop 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "width": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(253, 232, 128, 1)",
          "borderColor": "#231F20",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 30,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#231F20",
            "borderRadius": 9,
            "height": 18,
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly with disable 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "width": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(53, 53, 53, 0.1)",
          "borderColor": "rgba(53, 53, 53, 0.5)",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 30,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(53, 53, 53, 0.5)",
            "borderRadius": 9,
            "height": 18,
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;
